### PR TITLE
Modifying buildkit print statement

### PR DESCRIPTION
**Pull request by taytzehao**
Added a function to modify the print statement that originates from the buildkit. This allows us to replace or even reformat certain texts that may not be the clearest.

### CODE REVIEW

**Comment by alexcb** [comment]
relates to #3183

File: outmon/solvermon.go [hunk 1]
```diff
@@ -413,3 +414,14 @@ func (sm *SolverMonitor) reprintFailure(errVertex *vertexMonitor, phaseText stri
 	}
 	errVertex.printError()
 }
+
+
+func (sm *SolverMonitor) modifyId(id string) string {
+	modifiedId := id
+	for old, new := range map[string]string{
+		"transferring docker.io/library":  "transferring locally to docker.io/library",
```
Comment (alexcb) [inline outmon/solvermon.go:422]: this isn't generic, how will this help with other image names?

e.g.

```
./earthly ./examples/python+docker
```

produces

```
              output | [----------] 100% transferring docker.io/earthly/examples:python
```

File: outmon/solvermon.go [hunk 2]
```diff
@@ -413,3 +414,14 @@ func (sm *SolverMonitor) reprintFailure(errVertex *vertexMonitor, phaseText stri
 	}
 	errVertex.printError()
 }
+
+
+func (sm *SolverMonitor) modifyId(id string) string {
+	modifiedId := id
+	for old, new := range map[string]string{
```
Comment (alexcb) [inline outmon/solvermon.go:421]: redefining this map on each call seems inefficient, it should be global and created once via go's `init()` function.

**Review by alexcb — CHANGES_REQUESTED** [review]
thanks for tackling this; just a few minor issues.

File: outmon/solvermon.go (hunk 1 above)
Comment (taytzehao) [reply outmon/solvermon.go:422]: I think there are multiple criterias we have to consider if we want to determine it is transfering locally or remotely. We need to determine if the user's docker is logged in, destination username is the same as the username of the user, then only we can determine if the image would be push locally or remotely @alexcb. Let me know your thoughts. 

Of course, this is assuming that `SAVE IMAGE` only works with docker hub and not other container image repository. If there are other repository type, we may need to identify the different types of syntax

Comment (alexcb) [reply outmon/solvermon.go:422]: I agree -- it should work in a way that's generic to all these use cases.

**Review by vladaionescu — COMMENTED** [review]
Note that the whole solvermon logic is being replaced by logstream very soon.

**Comment by alexcb** [comment]
Closing this, due to this codepath no longer being relevant* since we have enabled the logstream (aka logbus) feature by default.

* It's only relevant to users who go out of their way to disable the logstream feature (to revert back to the old solvermon code). Long term we will delete solvermon code.

**Pull request closed without merging**